sender/sql.go

```diff
@@ -125,6 +132,21 @@ func (sq *SQL) init() {
 	sq.prep()
 }
 
+func verifySQLiteConn(file string) bool {
+	fd, err := os.Stat(file)
+
+	if os.IsNotExist(err) || fd.IsDir() {
+		return false
+	}
+
+	// Check if user has RW permissions to file. Both required.
+	if ((fd.Mode() & 0b010000000) != 0b010000000) || ((fd.Mode() & 0b100000000) != 0b100000000) {
```

Comment:
Consider if this can be made more readable by using e.g. the `io` package, such as [`fs.FileMode`](https://pkg.go.dev/io/fs#FileMode) and some bit-magic.

Furthermore, this checks if _any user_ has rw access to the file (here: root), not necessarily skogul (or the user running skogul), which results in the following:
```
$ stat --format '%a | %A %U:%G %n' /tmp/skogul.sqlite
644 | -rw-r--r-- root:wheel /tmp/skogul.sqlite
$ go run cmd/skogul/main.go cmd/skogul/man.go -f conf.json
ERRO[0000] Failed to transform and send metrics: sending metrics failed: sender failed: attempt to write a readonly database  category=receiver receiver=tester
```
(this should have failed in the init call, not be allowed to proceed on to "send".)
(sender config used:
```json
{
  "type": "sql",
  "driver": "sqlite3",
  "query": "INSERT INTO test VALUES(${timestamp},${metadata.src},${name},${data});",
  "connstr": "/tmp/skogul.sqlite"
}
```
)

Furthermore, on the contrary, it disallows skogul from writing to the database file if the _owner_ of the file does not have read or write access, but skogul does (notice the `rw-` for 'others', but not the owner (root)):
```
$ stat --format '%a | %A %U:%G %n' /tmp/skogul.sqlite
466 | -r--rw-rw- root:wheel /tmp/skogul.sqlite
$ go run cmd/skogul/main.go cmd/skogul/man.go -f conf.json
ERRO[0000]                                               category=sender driver=sqlite3 error="failed to initialize. Verify that /tmp/skogul.sqlite file exists or has correct permissions" sender=sql
ERRO[0000] Failed to transform and send metrics: sending metrics failed: sender failed: database initialization failed: failed to initialize. Verify that /tmp/skogul.sqlite file exists or has correct permissions  category=receiver receiver=tester
```
(I first thought of this as if skogul for some reason did _not_ have `rw-` on the file, but either `group` or `others` _did_, in which case I _think_ skogul still would have access? Which would be an extreme edge case. But then I realised that this applies anyways, because it might not be skogul _owning_ file file, even though it has read/write access - e.g. if it is owned by another system user, but skogul will read/write to it.)

Reply:
We went through this together and Kamil removed most of the check, but left the basic os.Stat() check and fd.IsDir, this is mainly to prevent panics, but everything else is caught by db.Open() now, and seems to report correctly. One last look, @sklirg ?

Reply:
LGTM 🎉